src/index/ivf/ivf.cc

```diff
@@ -273,6 +288,17 @@ IvfIndexNode<T>::Train(const DataSet& dataset, const Config& cfg) {
             index = std::make_unique<faiss::IndexIVFPQ>(qzr, dim, nlist, ivf_pq_cfg.m.value(), nbits, metric.value());
             index->train(rows, (const float*)data);
         }
+        if constexpr (std::is_same<faiss::IndexScaNN, T>::value) {
+            const ScannConfig& scann_cfg = static_cast<const ScannConfig&>(cfg);
+            auto nlist = MatchNlist(rows, scann_cfg.nlist.value());
+            bool is_cosine = base_cfg.metric_type.value() == metric::COSINE;
+            qzr = new (std::nothrow) typename QuantizerT<T>::type(dim, metric.value());
```

Comment:
Prefer to use smart pointers here. Any good reason for not using?

Reply:
consistent with current usage..